### DISCUSSION TITLE
Hide zero-length sliders

### DISCUSSION THREAD

**Discussion by Wieku** [ppy/osu#13139] [Ideas]
**Describe the new feature:** 
Some loved maps (like [this](https://osu.ppy.sh/beatmapsets/884669#osu/1935058), [this](https://osu.ppy.sh/beatmapsets/791900#osu/1676490) (not really loved but a bit popular map) and [this](https://osu.ppy.sh/beatmapsets/832364#osu/1743837)) use a gimmick that when slider's pixelLength is set to <= 0 it completely disappears from playfield but its head is normally clickable. Map creators use this to replace them with storyboarded hitcircles or other shapes. I think It would be nice to add the same behavior in lazer.

**Comment by peppy** [comment]
In such cases, how does the slider behave different from a hitcircle? Describing the full behaviour here may help us decide whether we can support this bug abuse or not.

**Comment by abraker95** [comment]
taken from here https://osu.ppy.sh/community/forums/topics/801973

>All sliders have, by default, a different hitwindow. It's normal to hit the sliderhead and receive 300s, despite having clicked on them during the 100/50 hitwindow. Clicking way too early on a sliderhead also results in a "miss", breaking your combo and not registering your click.
>
>However, invisible sliders don't have a normal hitwindow.
If you click way too early into them, it works the same way as a slider.
However, after the +0ms mark (The precise time the hitobject is on the map), it gives a miss instantaneously. 

But I think this isn't desired behaviors since it's in the "flaws of this technique" section. There are also nuances with hitsounding; sliderhead plays at same time as slidertail.

In all, I think what is actually desired is the ability to create invisible hitobjects.

**Comment by peppy** [comment]
So the hitwindow is already kinda weird to support as sliders in lazer have different lenience. Making them invisible is probably less of an issue.

**Comment by Wieku** [comment]
No no, changing the hit window behavior is out of question. Actually current lazer behavior is better, because those "hit circles" are clickable as normal hitcircles (even after slider's end time) with normal judgements. Only change needed is to hide them in such cases.

**Comment by GeNeRaL-ShAdOw** [comment]
Yes. If we listen carefully, for the map for aleph 0 by Mr Helix, at point 00:28:623 - to point 00:40:409 - it has double hitsounds for each object. The sliders are just invisible, they are still there. https://osu.ppy.sh/beatmapsets/884669#osu/1935058
I'm sorry if I just stated the obvious. 

It would be cool to see it in osu!lazer though. It will continue to give rise to new opportunities even in lazer!

That being said, if we used a different type of hitobject in mapping/editor it would be useless unless there is a Storyboard backing it up. Maybe use a toggle for switching it on and off under advanced options. Just a suggestion.

**Comment by LittleEndu** [comment]
Is there any plans to support this? You could just have "invisible hitobject" be a new thing in lazer beatmap format and have the legacy beatmap converter make all 0 length sliders into invisible hit circles. 

And considering the use case (replacing invisible hitobjects with storyboarded ones) displaying a notice if the beatmap is started with storyboard disabled, something like "This beatmap has invisible hitobjects. It's probably meant to be played with the storyboard enabled.", would also be a good idea

**Reply by smoogipoo** [reply]
There's no immediate plans but I'd be willing to consider it if someone implemented it.
